### DISCUSSION TITLE
Domoticz temperature sensor to OTGW

### DISCUSSION THREAD

**Discussion by ocmeros** [rvdbreemen/OTGW-firmware#118] [Q&A] [answered]
Recently i've bought a new OTGW from the Nodo-shop. I've added my own Wemos D1 mini. And installed a MQTT broker to connect the OTGW to Domoticz.
In the Wiki I see there is a way to use the MQTT set command. I would like to use this for sending an outdoor temperature sensor from Domoticz to my thermostat. Would that be possible? And how can I set this up?

I hope someone can help me with this.

Remco

**Comment by rvdbreemen** [comment]
Welcome to the community...

Read the wiki: 
https://github.com/rvdbreemen/OTGW-firmware/wiki/Using-DSB1820-sensor-with-ESP-firmware

If you need help, you can best join the discord.

**Comment by rlagerweij** [comment]
For future reference, the procedure is describer here:

[https://github.com/rvdbreemen/OTGW-firmware/wiki/How-to-send-commands-to-the-OTGW-with-MQTT-or-REST-or-Net2Ser](https://github.com/rvdbreemen/OTGW-firmware/wiki/How-to-send-commands-to-the-OTGW-with-MQTT-or-REST-or-Net2Ser)

To quote:

"So sending Outside Temperature to your OTGW with a local measurement of the outside temperature then it is this: topic: 
OTGW/set/otgw-1234567890AB/command value: OT=21.5"

**Comment by ajongen** [comment]
I managed to sent the Outside Temperature from Domoticz (that is the temperature values from a Buienradar plugin) to OTGW using an Device Changed Event. For this I adapted the info found on:
https://bitbucket.org/frankiepankie/domoscripts/src/master/lua/script_device_outsidetemp_otmonitor.lua?at=master&fileviewer=file-view-default

It basically constructs the http://192.168.178.xx/command?OT=value command that is sent to the otmonitor.

Now I would like to do the same for the "Return water temperature". My OTGW is located away from my boiler so I do not have the possibility to attach the temperature sensor to the GPIO of the board. But I have a sensor that feeds its data into Domoticz. 

Is there a way to send the Return water temperature from a Domoticz sensor to otmonitor in a simular way as can be done with the outside temperature?
Are there any other values that can be set that way?

Thanks for the help on this!

Cheers, Armand

**Reply by rlagerweij** [reply]
It could be that your boiler is capable of sending the return temperature but just doesn't do so by default. 

You can check if this is the case by  connecting through telnet into your OTGW and send the "PM=28" command. (without the quotes) If you get a response with a temperature you can send the "AA=28" command to receive it regularly.

**Reply by ajongen** [reply]
Unfortunatelly my boiler isn't able to sent the return temperature (Intergas Kombi Kompakt HR 36/30 A). This is in line with the information on http://otgw.tclcode.com/matrix.cgi#boilers

I tested the PM=28 without success. But I have a sensor measuring the temperature and feeding it into my Domoticz. So I have the data. But how to get this fed into otmonitor, simular to what is possible with the outside temperature (OT=value command)?

**Comment by rlagerweij** [comment, marked as answer]
To my knowledge the OTGW does not have a command to set the return temperature. I'm also not quite sure what the use of sending this data would be? I see the use for outdoor temperature but I don't think any boiler would allow overriding a parameter like that which is so essential to its own operation.
If however, you are doing it to use some sort of function or plotting capability in Domotics, maybe you could try to just send the sensor data to the mqtt topic that the OTGW firmware has created. This would make Domoticz think it was reported by the OTGW.

**Reply by rvdbreemen** [reply]
@ocmeros you could if you like use an external DS18B20 temperature sensor to record the temperature on your return flow to your boiler. This can then be used in your Home Automation solution to graph / record it. 

More information here: https://github.com/rvdbreemen/OTGW-firmware/wiki/Using-DS18B20-sensor-with-ESP-firmware

**Reply by ajongen** [reply]
> 

@rlagerweij Indeed, I probably take this to lightly. I was not aware that sending this information to the OTGW/otmonitor would also sent it to the boiler who might have an internal value that could be overwritten and skrew up the modulation. My goal was just to also see the retutn water temperature in the graphs of otmonitor. I apparently need to do that in some other environment like Domoticz. Thanks for the insight!
